libs/datasets/combined_datasets.py

```diff
@@ -88,25 +91,33 @@ def SOURCE_TYPE(self):
         """Implements the same interface as the wrapped DataSource class."""
         return self.data_source_cls.SOURCE_TYPE
 
+    @staticmethod
+    def _get_location_ids(region_mask_or_regions: RegionMaskOrRegions) -> Collection[str]:
+        if isinstance(region_mask_or_regions, RegionMask):
+            return region_mask_to_location_ids(region_mask_or_regions)
+        elif isinstance(region_mask_or_regions, Region):
+            return [region_mask_or_regions.location_id]
+        else:
+            location_ids = []
+            for mask_or_region in region_mask_or_regions:
+                if isinstance(mask_or_region, RegionMask):
+                    location_ids.extend(region_mask_to_location_ids(mask_or_region))
+                else:
+                    location_ids.append(mask_or_region.location_id)
```

Comment:
nit: You could probably replace the above 4 lines with a recursive call?

Reply:
Or changing `RegionMaskOrRegions` to be only `Collection[Union[Region, RegionMask]]`, which is just a convenience so we don't need to wrap single regions in a list with one element.